log/testutil.go

```diff
@@ -0,0 +1,122 @@
+package log
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+// FieldValidator facilitates testing of Span.Log*() implementations.
+//
+// FieldValidator SHOULD ONLY BE USED FOR TESTS. It is not intended for
+// production code.
+//
+// Usage:
+//
+//     fv := log.NewFieldValidator(t, someLogStructure.Fields)
+//     fv.
+//         ExpectNextFieldEquals("key1", reflect.String, "some string value").
+//         ExpectNextFieldEquals("key2", reflect.Uint32, "4294967295")
```

Comment:
I like this approach more than `ExpectNextFieldEquals("key2", uint32(4294967295))` because `reflect.Interface` is an important case to test with object logging.